packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker/hooks.ts

```diff
@@ -0,0 +1,78 @@
+import { RefObject, useEffect, useState } from 'react';
+
+import { TimeOption } from '@grafana/data';
+
+const modulo = (a: number, n: number) => ((a % n) + n) % n;
```

Comment:
Can you explain the logic behind this?

Reply:
This was a direct copy of how it is done in the `Menu` [hook](https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/Menu/hooks.ts#L4). From what I understand this is a calculation for when a number goes "out of bounds" and you want to loop back to the top/bottom of the list, for instance if you are moving from index 7 to index 8, `modulo(8,8)` returns `0` so it will select the first item.